pkg/config/constants/constants.go

```diff
@@ -133,4 +133,7 @@ const (
 
 	// TrustworthyJWTPath is the defaut 3P token to authenticate with third party services
 	TrustworthyJWTPath = "./var/run/secrets/tokens/istio-token"
+
+	// DryRunAnnotation enables dry-run mode of a resource, currently only AuthorizationPolicy is supported.
+	DryRunAnnotation = "istio.io/dry-run"
```

Comment:
After reading the doc, I am not sure why an annotation rather than a field in authz policy is prefered. It seems the primary benefit is that we can avoid full API review, which I don't think we should do anyways.?

The other benefit listed is about using it for other Istio API types. It seems unlikely the any other API can support this in the near future, but even if it did I would expect it to follow the same pattern as `exportTo` - a field shared by many types, but defined in each API explicitly.

In many ways, using a generic annotation should actually raise the level of scrutiny as we are presenting this as a generic "istio dry run" rather than "authorization policy dry run".

Reply:
Sent out a PR: https://github.com/istio/api/pull/1933